sklearn/plot/tests/test_heatmap.py

```diff
@@ -0,0 +1,114 @@
+from sklearn.plot import plot_heatmap
+from sklearn.plot import plot_confusion_matrix
+from sklearn.plot import plot_gridsearch_results
+from sklearn.datasets import load_iris
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
+from sklearn.utils.testing import SkipTest
+from sklearn.utils.testing import assert_raises
+import numpy as np
+
+
+def test_heatmap():
+    try:
+        import matplotlib
+    except ImportError:
+        raise SkipTest("Not testing plot_heatmap, matplotlib not installed.")
+
+    import matplotlib.pyplot as plt
+
+    with matplotlib.rc_context(rc={'backend': 'Agg', 'interactive': False}):
+        plt.figure()
+        rng = np.random.RandomState(0)
+        X = rng.normal(size=(10, 5))
+        # use mixture of default values and keyword args
+        plot_heatmap(X, ylabel="y-axis",
+                     xticklabels=["a", "b", "c", "d", "efgh"],
+                     cmap="Paired", ax=plt.gca())
+
+        plt.draw()
+        plt.close()
+
+
+def test_confusion_matrix():
+    try:
+        import matplotlib
+    except ImportError:
+        raise SkipTest("Not testing plot_heatmap, matplotlib not installed.")
+
+    import matplotlib.pyplot as plt
```

Comment:
because matplotlib is being used in the line below "with matplotlib.rc_context(rc={'backend': 'Agg', 'interactive': False}):"
i took this from your original test function. will this take longer to run?